src/web/lib/decideLogo.ts

```diff
@@ -0,0 +1,33 @@
+import { Design, Special, Pillar } from '@guardian/types';
+
+const shouldUseLogoForDarkBackground = (format: Format): boolean => {
+	if (format.theme === Special.SpecialReport) return true;
+	switch (format.design) {
+		case Design.Media:
+			return true;
+		case Design.LiveBlog:
+			switch (format.theme) {
+				case Special.Labs:
+					return false;
+				case Pillar.News:
+				case Pillar.Sport:
+				case Pillar.Opinion:
+				case Pillar.Lifestyle:
+				case Pillar.Culture:
+				default:
+					return true;
+			}
+		default:
+			return false;
+	}
+};
+
+export const decideLogo = (
+	format: Format,
+	branding: Branding,
+): BrandingLogo => {
+	return shouldUseLogoForDarkBackground(format) &&
+		branding.logoForDarkBackground
+		? branding.logoForDarkBackground
+		: branding.logo;
```

Comment:
Could this be shortened to `... && (branding.logoForDarkBackground || branding.logo)`?

Reply:
I think it could be shortened, but I think the braces would be like `(... && ...) || ...` instead - which probably indicates why we might prefer the ternary still because it is harder to get wrong and easier to grep IMO.